src/connectionTool/Main.java

```diff
@@ -0,0 +1,212 @@
+        ConfigArguments configArguments = new ConfigArguments();
+        JCommander jc = JCommander.newBuilder()
+                .addObject(helpArgument)
+                .addCommand(detailsArgument)
+                .addCommand(configArguments)
+                .build();
+        if (args.length == 0){
+            jc.usage();
+            System.exit(0);
+        }
+        try {
+            jc.parse(args);
+        }catch (ParameterException e){
+            System.out.println(e.getMessage());
+            LogSaver.appendLog(e.toString());
+            System.exit(0);
+        }
+        String parsedCmdStr = jc.getParsedCommand();
+        ConnectionMode connectionMode = null;
+        if (helpArgument.isHelp()){
+            jc.usage();
+            System.exit(0);
+        }
+        switch (parsedCmdStr) {
+            case "details":
+                connectionMode = ConnectionMode.DETAILS;
+                break;
+
+            case "config":
+                connectionMode = ConnectionMode.CONFIG;
+                break;
+
+            default:
+                System.err.println("Invalid command: " + parsedCmdStr);
+                System.exit(0);
+        }
+        if (connectionMode == ConnectionMode.DETAILS){
+            if (detailsArgument.getHelp()){
+                jc.usage();
+                System.exit(0);
+            }
+            ConnectionCheck connectionCheck = new ConnectionCheck(detailsArgument.getConnectionString(),
+                    detailsArgument.getUsername(),
+                    detailsArgument.getPassword(),
+                    detailsArgument.getTimeout(),
+                    detailsArgument.isSsl(),
+                    detailsArgument.isTrustCertificates());
+            ping(connectionCheck.getHost(), connectionCheck.getTimeout());
+            connect(connectionCheck.getTimeout(), detailsArgument.getDriverPath(),
+                    connectionCheck.getProvider(),
+                    connectionCheck.getConnectionString(),
+                    connectionCheck.getProperties());
+        }
+        else {
+            if (configArguments.getHelp()){
+                jc.usage();
+                System.exit(0);
+            }
+            makeConnection(configArguments.getConfigPath(), configArguments.getDriverPath());
+        }
+    }
+    private static void ping(String hostName, int timeout){
+        boolean isReachable = false;
+        try {
+            isReachable = InetAddress.getByName(hostName).isReachable(timeout * 1000);
+        } catch (final UnknownHostException e) {
+            LogSaver.appendLog(e.toString());
+            System.out.println("Host is unknown");
+            System.exit(0);
+        } catch (IOException e) {
+            System.out.println("Couldn't ping host, check logs for details!");
+            LogSaver.appendLog(e.toString());
+            System.exit(0);
+        }
+        if (isReachable) {
+            System.out.println("Host is reachable");
```

Comment:
it is a common pattern that you log to file & print to console at the same time. I think you could extract it to a separate method that takes a message as argument